Fix incorrect uses of `esc_attr*` and `esc_html*` picked up by WPORG checks

Aljullu opened the pull request:
This PR fixes the usage of several escaping functions according to a list of warnings that appears when uploading the plugin to WordPress.org.

### How to test the changes in this Pull Request:
I think changes are quite safe, but this PR can be tested checking the templates/flows listed below and ensuring all texts appear in the screen without issues:
* Sticky add to cart bar in the product page.
* Comment template + 'comment awaiting moderation' template when adding a comment to a post.
* Primary navigation (top menu).
* Skip links (triggered when navigating the page with the <kbd>Tab</kbd> key).
* Install/welcome flow (triggered when installing Storefront for the first time).

### Changelog

> Tweak – Fix incorrect uses of `esc_attr*` and `esc_html*` picked up by WPORG checks.

Aljullu committed:
Use correct escaping functions

haszari commented:
Adding the screenshot of the WPORG scan results for posterity & context:

![Screenshot_2020-11-10 Submit Your Theme or Theme Update to the Directory WordPress org](https://user-images.githubusercontent.com/4167300/98995348-13a78580-2596-11eb-8402-18bf6a063338.png)

haszari reviewed (approved):
Changes look 👌  - nice that WPORG caught these, it's clear we were misusing `esc_attr` & `esc_html`. 

I tested all the flows, including removing WooCommerce and deleting `storefront_nux_dismissed` option to trigger the `learn more` text. Everything worked correctly.

@Aljullu I think we should keep an eye on the WPORG warnings when we do a release. I'm not sure why I missed these in the past. If you like, feel free to add a line to the release doc to remind us to check the output of the WPORG upload checks, and log any issues if needed.

haszari commented:
I updated the title and changelog to so it's clearer / more specific.

Aljullu commented:
Thanks for the review @haszari!

> @Aljullu I think we should keep an eye on the WPORG warnings when we do a release. I'm not sure why I missed these in the past. If you like, feel free to add a line to the release doc to remind us to check the output of the WPORG upload checks, and log any issues if needed.

I added this step and pushed to `trunk` directly: 8534b4f0dfc88dc591e032d6a7c90cb5f79a4eb1. :slightly_smiling_face: 

> I updated the title and changelog to so it's clearer / more specific.

Thank you!
